src/functional/fencil_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -42,7 +43,116 @@
 from functional.iterator import ir as itir
 
 
+_vertical_dimension = "unstructured::dim::vertical"
+_horizontal_dimension = "unstructured::dim::horizontal"
+
+
+def _get_domains(closures: Iterable[itir.StencilClosure]) -> Iterable[itir.FunCall]:
+    domains = [c.domain for c in closures]
+    assert all(isinstance(d, itir.FunCall) for d in domains)
+    return domains
+
+
+def _extract_grid_type(domain: itir.FunCall) -> common.GridType:
+    if domain.fun == itir.SymRef(id="cartesian_domain"):
+        return common.GridType.CARTESIAN
+    else:
+        assert domain.fun == itir.SymRef(id="unstructured_domain")
+        return common.GridType.UNSTRUCTURED
+
+
+def _get_gridtype(closures: list[itir.StencilClosure]) -> common.GridType:
+    domains = _get_domains(closures)
+    grid_types = set(_extract_grid_type(d) for d in domains)
```

Comment:
```suggestion
    grid_types = {_extract_grid_type(d) for d in domains}
```

Reply:
If this is the only other change, I would put it in the follow up pr mentioned above.